sale_stock_secondary_unit/models/stock_rule.py

```diff
@@ -0,0 +1,39 @@
+# Copyright 2019 Tecnativa - Sergio Teruel
+# License AGPL-3.0 or later (https://www.gnu.org/licenses/agpl).
+from odoo import models
+
+
+class StockRule(models.Model):
+    _inherit = "stock.rule"
+
+    def _get_stock_move_values(
+        self,
+        product_id,
+        product_qty,
+        product_uom,
+        location_id,
+        name,
+        origin,
+        company_id,
+        values,
+    ):
+        res = super(StockRule, self)._get_stock_move_values(
+            product_id,
+            product_qty,
+            product_uom,
+            location_id,
+            name,
+            origin,
+            company_id,
+            values,
+        )
+        if values.get("sale_line_id", False):
```

Comment:
Default is 'None' and is evaluated as falsy value too
```suggestion
        if values.get("sale_line_id"):
```